### PR TITLE
Log successful AcceptBlockHeader()

### PR DESCRIPTION
**Pull request by Sjors**
Knowing when a header was first seen may help distinguish between a regular reorg and a selfish mining or eclipse attack.

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -3828,6 +3828,7 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
         return state.Invalid(BlockValidationResult::BLOCK_HEADER_LOW_WORK, "too-little-chainwork");
     }
     CBlockIndex* pindex{m_blockman.AddToBlockIndex(block, m_best_header)};
+    LogPrint(BCLog::VALIDATION, "%s: added new block header %s\n", __func__, hash.ToString());
```
Comment (dergoegge) [inline src/validation.cpp:3831]:
```suggestion
    LogPrint(BCLog::VALIDATION, "added new block header %s\n", hash.ToString());
```

We have an option for logging source locations `-logsourcelocations`.

**Review by dergoegge — COMMENTED** [review]
Concept ACK

File: src/validation.cpp (hunk 1 above)
Comment (Sjors) [reply src/validation.cpp]: Done!

**Comment by jamesob** [comment]
Beat me to it! I was just about to open this PR :).

For what it's worth, I think that the right approach here is to make this message `LogPrintf()` and omit it when in initial block download, the rationale being that if something weird happens on the network (like the inspiration for this PR last night), having a lot of data available to investigate would be nice, and most people don't run with verbose logging. Since the additional log volume here is quite minimal (one extra line per block, basically), I don't think there's much cost.

**Comment by Sjors** [comment]
I opened #27277 to make `-debug=validation` less noisy, making it a more attractive setting. At least _after_ IBD, since it produces ~4 lines per block: AcceptBlockHeader, NewPoWValidBlock, Pre-allocating and BlockChecked.

**Comment by fanquake** [comment]
cc @0xB10C

File: src/validation.cpp [hunk 2]
```diff
@@ -3828,6 +3828,7 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
         return state.Invalid(BlockValidationResult::BLOCK_HEADER_LOW_WORK, "too-little-chainwork");
     }
     CBlockIndex* pindex{m_blockman.AddToBlockIndex(block, m_best_header)};
+    LogPrint(BCLog::VALIDATION, "added new block header %s\n", hash.ToString());
```
[the remark this answers is not in the document]
Comment (Sjors) [reply src/validation.cpp]: Done

**Comment by Sjors** [comment]
At this point I prefer #27278, but I'll leave this open in case that PR doesn't make it. This one is (slightly) easier to review.

**Pull request closed without merging**